jobs/kpi-forecasting/kpi-forecasting/models/prophet_forecast.py

```diff
@@ -0,0 +1,119 @@
+            mcmc_samples=0,
+        )
+
+        if self.use_holidays:
+            self.model.add_country_holidays(country_name="US")
+
+        # Modify observed data to have column names that Prophet expects, and fit
+        # the model
+        self.model.fit(self.observed_df.rename(columns=self.column_names_map))
+
+    def _predict(self) -> pd.DataFrame:
+        """
+        Forecast using `self.model`.
+        """
+        # generate the forecast samples
+        samples = self.model.predictive_samples(
+            self.dates_to_predict.rename(columns=self.column_names_map)
+        )
+        df = pd.DataFrame(samples["yhat"])
+
+        # add dates_to_predict to the output
+        df["submission_date"] = self.dates_to_predict
+
+        return df
+
+    def _predict_legacy(self) -> pd.DataFrame:
+        """
+        Recreate the legacy format used in
+        `moz-fx-data-shared-prod.telemetry_derived.kpi_automated_forecast_v1`.
+        """
+        # TODO: This method should be removed once the forecasting data model is updated:
+        # https://docs.google.com/document/d/18esfJraogzUf1gbZv25vgXkHigCLefazyvzly9s-1k0.
+
+        df = self.model.predict(
+            self.dates_to_predict.rename(columns=self.column_names_map)
+        )
+
+        # set legacy column values
+        if "dau" in self.metric_hub.alias.lower():
+            df["metric"] = "DAU"
+        else:
+            df["metric"] = self.metric_hub.alias
+
+        df["forecast_date"] = str(datetime.utcnow().date())
+        df["forecast_parameters"] = str(
+            json.dumps({**self.parameters, "holidays": self.use_holidays})
+        )
+
+        if "desktop" in self.metric_hub.alias.lower():
+            df["target"] = "desktop"
+        elif "mobile" in self.metric_hub.alias.lower():
+            df["target"] = "mobile"
+        else:
+            df["target"] = None
+
+        columns = [
+            "ds",
+            "trend",
+            "yhat_lower",
+            "yhat_upper",
+            "trend_lower",
+            "trend_upper",
+            "additive_terms",
+            "additive_terms_lower",
+            "additive_terms_upper",
+            "extra_regressors_additive",
+            "extra_regressors_additive_lower",
+            "extra_regressors_additive_upper",
+            "holidays",
+            "holidays_lower",
+            "holidays_upper",
+            "regressor_00",
+            "regressor_00_lower",
+            "regressor_00_upper",
+            "weekly",
+            "weekly_lower",
+            "weekly_upper",
+            "yearly",
+            "yearly_lower",
+            "yearly_upper",
+            "multiplicative_terms",
+            "multiplicative_terms_lower",
+            "multiplicative_terms_upper",
+            "yhat",
+            "target",
+            "forecast_date",
+            "forecast_parameters",
+            "metric",
+        ]
+
+        for column in columns:
```

Comment:
is this expected and intended behaviour? Or does one of these columns missing a sign of something bad?

Reply:
This is expected behavior in the "legacy" framework -- mobile and desktop forecasts returned different numbers of columns (example `"regressor_00"` [here](https://github.com/mozilla/docker-etl/blob/cdc52b62b4ef7150597733ba03829bf050b28e99/jobs/kpi-forecasting/kpi-forecasting/Utils/FitForecast.py#L33)). Most of these columns are not being used (hence the project to revamp the data model), but I left the logic in place for consistency during the transition.